lib/PreviewProvider.js

```diff
@@ -237,13 +279,17 @@ PreviewProvider.prototype = {
       let response = yield this._asyncGetLinkData(linkURLs);
       if (response.ok) {
         let responseJson = yield response.json();
+        let currentTime = Date.now();
         newLinks.forEach(link => {
           const data = responseJson.urls[link.sanitizedURL];
           if (!data) {
             return;
           }
-          ss.storage.embedlyData[link.cacheKey] = data;
-          ss.storage.embedlyData[link.cacheKey].accessTime = Date.now();
+          ss.storage.embedlyData[link.cacheKey] = Object.assign({}, ss.storage.embedlyData[link.cacheKey], data);
+          if (updateAccessTime) {
+            ss.storage.embedlyData[link.cacheKey].accessTime = currentTime;
+          }
+          ss.storage.embedlyData[link.cacheKey].refreshTime = currentTime;
```

Comment:
was gonna ask about `accessTime` vs `refreshTime` but the tests make it obvious what's they are for 👍